common/src/stack/report-system/command/report/system/tests/test_other_dhcp.py

```diff
@@ -57,7 +57,27 @@ def test_other_dhcp_server(interface):
 	# Set up socket with passed in interface
 	send_dhcp = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
 	send_dhcp.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-	send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	# Bail if we can't setup a socket on the current interface
+	try:
+		send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	except OSError:
+		pytest.skip(f'Could not setup socket for {interface}')
+
+	# If the MAC address isn't in the Stacki database, try elsewhere
+	if not pxe_interfaces[interface]:
+		find_mac = _exec(['cat'] + [f'/sys/class/net/{interface}/address'])
```

Comment:
`find_mac = open('...').read()` is an alternative here, or some similar construct. No need for forking and pipes.

Reply:
That's a good suggestion, I believe @bsanders had a similar idea to use `read()`